erpnext/stock/doctype/stock_entry/test_stock_entry.py

```
@@ -2410,6 +2410,51 @@ def test_validation_as_per_bom_with_continuous_raw_material_consumption(self):
+		Unit test case to check the document naming rule with company condition
+		For Quality Inspection, when created from Stock Entry.
+		"""
+		from erpnext.accounts.report.trial_balance.test_trial_balance import create_company
+		from erpnext.controllers.stock_controller import make_quality_inspections
+		from erpnext.stock.doctype.warehouse.test_warehouse import create_warehouse
+
+		# create a separate company to handle document naming rule with company condition
+		qc_company = create_company(company_name="Test Quality Company")
+
+		# create document naming rule based on that for Quality Inspection Doctype
+		qc_naming_rule = frappe.new_doc(
```

Ruff (0.15.2)

[error] `frappe` may be undefined, or defined from star imports

(F405)